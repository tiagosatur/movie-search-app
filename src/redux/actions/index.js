export * from './movies';
export * from './details';