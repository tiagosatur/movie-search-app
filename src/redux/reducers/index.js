import { combineReducers } from 'redux'
import {movie} from './movie'
import {details} from './details'
import {user} from './user'

export default combineReducers ({
    movie,
    details,
    user,
})