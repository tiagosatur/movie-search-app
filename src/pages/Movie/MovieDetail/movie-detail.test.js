import React from 'react';
import {  Provider } from 'react-redux'
import {createStore, applyMiddleware} from 'redux';
import thunk from 'redux-thunk';
import rootReducer from '../../../redux/reducers'
import { mount, shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import { MemoryRouter } from "react-router";

//import { initialState } from '../../../redux/reducers/details';
import ConnectedMovieDetail, { MovieDetail } from './MovieDetail';
import movieDetailsApiMock from '../../../mocks/movieDetailsApiMock'

describe('MovieDetails Page tests', () => {
    const initialState = {
        pending: false,
        details: movieDetailsApiMock,
        error: null
    }
    let store = createStore(rootReducer, initialState, applyMiddleware(thunk));

    const wrapper =  mount(
        <Provider store={store}>
            <MemoryRouter>
                <ConnectedMovieDetail />
            </MemoryRouter>
        </Provider>
    );

    it('should render the Movie Details', () => {
        expect(wrapper.find('.movie-details').exists()).toBe(true);
    });

    it('should call MovieDetails snapshot correctly', () => {
        const tree = renderer
            .create(wrapper)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
})
