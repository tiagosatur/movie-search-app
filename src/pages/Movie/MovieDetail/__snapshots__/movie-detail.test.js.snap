// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieDetails Page tests should call MovieDetails snapshot correctly 1`] = `
<div
  className="loading-spinner sc-htoDjs dkuhNH"
>
  <div
    className="sc-gzVnrw cbeMKQ"
  />
</div>
`;
